### PR TITLE
NXP-26485: fix invalidation of root document

### PR DESCRIPTION
**Pull request by efge**
T&P in progress

### CODE REVIEW

File: nuxeo-core/nuxeo-core-storage-sql/nuxeo-core-storage-sql-test/src/test/java/org/nuxeo/ecm/core/storage/sql/TestSQLBackend.java [hunk 1]
```diff
@@ -145,6 +145,39 @@ public void testRootNode() throws Exception {
...
+        ACLRow acl1 = new ACLRow(1, "test", true, "Write", "steve", null);
+        ACLRow acl2 = new ACLRow(1, "test", true, "Write", "jobs", null);
+        prop.setValue(new ACLRow[] { acl1, acl2 });
+        session.save(); // send invalidations, fragment in session2 is INVALIDATED_MODIFIED
+
+        // in session2 re-change acl
+        session2.save(); // fetch invalidations
+        Node root2 = session2.getNodeById(rootNodeId); // fetch root by id
+        CollectionProperty prop2 = root2.getCollectionProperty(Model.ACL_PROP);
+        prop2.setValue(new ACLRow[] { acl1 });
```
Comment (kevinleturc) [inline nuxeo-core/nuxeo-core-storage-sql/nuxeo-core-storage-sql-test/src/test/java/org/nuxeo/ecm/core/storage/sql/TestSQLBackend.java]: Should we assert before setting new acl that we have 2 acls ? in order to assert invalidations.

Comment (efge) [reply nuxeo-core/nuxeo-core-storage-sql/nuxeo-core-storage-sql-test/src/test/java/org/nuxeo/ecm/core/storage/sql/TestSQLBackend.java]: We know invalidations work in general. The last test in the method is the only special case that behaves incorrectly.

**Comment by kevinleturc** [comment]
👍

File: nuxeo-core/nuxeo-core-storage-sql/nuxeo-core-storage-sql-test/src/test/java/org/nuxeo/ecm/core/storage/sql/TestSQLBackend.java [hunk 2]
```diff
@@ -145,6 +145,39 @@ public void testRootNode() throws Exception {
...
 
+    /*
+     * The root node fragments must correctly be refetched when they are invalidated.
+     */
+    @Test
+    public void testRootNodeInvalidation() throws Exception {
+        Session session = repository.getConnection();
+        Serializable rootNodeId = session.getRootNode().getId();
+        Session session2 = repository.getConnection();
+        session2.getRootNode().getCollectionProperty(Model.ACL_PROP);
```
[the remark this answers is not in the document]
Comment (efge) [reply nuxeo-core/nuxeo-core-storage-sql/nuxeo-core-storage-sql-test/src/test/java/org/nuxeo/ecm/core/storage/sql/TestSQLBackend.java]: It allows to put in session2's cache the ACL fragments for the root node. These will be those that get invalidated later on.

[the remark this answers is not in the document]
Comment (ataillefer) [reply nuxeo-core/nuxeo-core-storage-sql/nuxeo-core-storage-sql-test/src/test/java/org/nuxeo/ecm/core/storage/sql/TestSQLBackend.java]: OK, might deserve a comment then.

[the remark this answers is not in the document]
Comment (efge) [reply nuxeo-core/nuxeo-core-storage-sql/nuxeo-core-storage-sql-test/src/test/java/org/nuxeo/ecm/core/storage/sql/TestSQLBackend.java]: Done

**Pull request merged**